generate-stats

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import ssl
+import sys
+import sqlite3
+import urllib.request
+import argparse
+import logging
+import re
+import json
+
+from task import github
+from machine.machine_core.constants import IMAGES_DIR
+
+def main():
+    parser = argparse.ArgumentParser(description='Store information about failed tests')
+    parser.add_argument("--db", default="cockpit-failures.db", help="Database name")
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose logging")
+    opts = parser.parse_args()
+
+    if opts.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+
+    db_conn = sqlite3.connect(opts.db)
+    cursor = db_conn.cursor()
+
+    all_tables = []
+
+    # Get all failures sorted by number of occurrences
+    top10 = []
+    rows = cursor.execute("select testname, COUNT(run) from Failures GROUP BY testname ORDER BY COUNT(run) DESC LIMIT 10;").fetchall()
+    for row in rows:
+        runs = cursor.execute("SELECT DISTINCT run from Failures WHERE testname = ? ORDER BY run DESC LIMIT 2;", (row[0], )).fetchall()
+        urls = cursor.execute("SELECT url from TestRuns WHERE id = ? or id = ?;", (runs[0][0], runs[1][0])).fetchall()
+        top10.append([row[0], row[1], [urls[0][0], urls[1][0]]])
+
+    all_tables.append({"name": "Top failures", "columns": ["Test name", "Count", "Logs"], "data": top10})
+
+    # Get failures by OS
+    oses = cursor.execute("SELECT DISTINCT context FROM TestRuns;").fetchall();
```

Comment:
let's call them context; for consistency, and it's also things like "-distropkg" or "/firefox"